### ISSUE TITLE
Stacking Context / Modals / Toast Notifs

### ISSUE THREAD

**Issue by chalabi2** [liftedinit/manifest-app#119]
**Issue**
Toast messages appear below modals that use Dialog element controls

**Synopsis**
There is a general stacking context issue for modals in this application. 

Daisy-ui has 3 modal opening methods 

1. Dialog element controls (recommended)
2. Invisible check-boxes
3. Anchor links

Option 1 pushes the modal to the "top-layer" this is custom CSS in daisy-ui that forces the modal to the topmost layer in the stacking context. This causes toast messages to appear below the modal regardless of the toast messages z-index. Even when you use a portal to force it to the top, daisy-ui's "top-layer" is still the highest in the stacking context. 

I believe option 2 suffers from the same issue and daisy-ui does not recommend using option 3. 

That leaves us with 2 options - 

In the token factory page under the `myDenoms` component you can see that we control the modals via `useState` implementing a `const [openXModal, setOpenXModal] = useState(false)` method. This fixes the toast message stacking context but puts the `sideNav` above the modal backdrop ( I believe this is fixable). 

OR 

Using headless/ui's `Dialog` methods which can be seen in the wallet modals. The reason why we have headless ui is because we pulled down cosmology-react examples for tailwindcss wallet modals and they were using headless/ui. But using headless UI in the `myGroups` component causes issues because we have to pass the state of the `const [openXModal, setOpenXModal] = useState(false)` through multiple components which also borks the stacking context. That issue is also fixable as Ive seen through testing. 

**TLDR** 

We have a stacking context issue that has multiple solutions but each one will require a minor change in how we open and close all modals.

**Comment by fmorency** [comment]
Thank you for the great description. Which option are you thinking about implementing?

**Comment by chalabi2** [comment]
> Thank you for the great description. Which option are you thinking about implementing?

I believe the first option is our best bet.

**Comment by chalabi2** [comment]
fixed in #127
